lib/reline/line_editor.rb

```diff
@@ -1148,6 +1153,29 @@ def input_key(key)
     modified
   end
 
+  def save_old_buffer
+    @old_buffer_of_lines = @buffer_of_lines.dup
+    @old_byte_pointer = @byte_pointer.dup
+    @old_line_index = @line_index.dup
+  end
+
+  MAX_PAST_LINES = 100
+  def save_past_lines
+    if @old_buffer_of_lines != @buffer_of_lines
+      if !@past_lines.empty? && @undoing
+        @past_lines.pop
```

Comment:
How about moving this step to `def undo`?
The current implementation looks like undo logic (== revert buffer_of_lines and revert past_lines) is separated into two parts.


In `def input_key`, this method can be used like this:
```ruby
save_past_lines unless @undoing # or maybe rename it to push_past_lines
@undoing = false
```

Reply:
Sounds good! I've fixed it.